refinery/analysis_manager/tasks.py

```diff
@@ -448,6 +531,53 @@ def _start_galaxy_analysis(analysis_uuid):
         _start_galaxy_analysis.update_state(state=celery.states.FAILURE)
 
 
+@task()
+def _tool_based_galaxy_file_import(analysis_uuid, file_store_item_uuid,
+                                   history_dict, library_dict):
+
+    analysis = _get_analysis(analysis_uuid)
+    analysis_status = _get_analysis_status(analysis_uuid)
+    tool = _get_tool(analysis_uuid)
+
+    connection = analysis.galaxy_connection()
+
+    file_store_item = FileStoreItem.objects.get(uuid=file_store_item_uuid)
+    library_dataset_dict = connection.libraries.upload_file_from_url(
+        library_dict["id"],
+        get_full_url(file_store_item.get_datafile_url())
+    )
+    history_dataset_dict = (
+        connection.histories.upload_dataset_from_library(
+            history_dict["id"],
+            library_dataset_dict[0]["id"]
+        )
+    )
+
+    tool.update_file_relationships_with_galaxy_history_data(
+        {
+            "refinery_file_uuid": file_store_item_uuid,
+            "galaxy_dataset_history_id": history_dataset_dict["id"]
+        }
+    )
+
+    number_of_files = len(tool.get_input_file_uuid_list())
+    single_file_percentage = (100 / number_of_files)
+    analysis_status.galaxy_import_progress = (
+        analysis_status.galaxy_import_progress + single_file_percentage
+    )
+    analysis_status.save()
+
+    if (analysis_status.galaxy_import_progress ==
```

Comment:
Testing floats for equality is never a good idea. Can you find some way to do this based on the number of files processed?